motor/motor_control/motorcontrol/motorcontrol.ino

```diff
@@ -0,0 +1,140 @@
+
+#include <ros.h>
+#include <ArduinoHardware.h>
+
+
+#include <ros.h>
+#include <std_msgs/Float64.h>
+#include <std_msgs/Bool.h>
+#include <tf/transform_broadcaster.h>
+#include <nav_msgs/Odometry.h>
+
+#include <ros/time.h>
+
+static unsigned long last_ros_count = 0;
+
+// ports (arbitrary numbers as placeholders)
+// right
+const int RFR =  4;      // digital; motor direction control; F/R == COM == 0 => anticlockwise; clockwise defined as forwards
+const int RSV = 5;       // controls speed; 5V = max speed
+// left
+const int LFR = 7;       // digital; motor direction control; F/R == COM == 0 => anticlockwise; clockwise defined as forwards
+const int LSV = 6;       // controls speed; 5V = max speed
+// commands
+const int light_pin = 13;  
+
+unsigned long current_ros = 0;
+
+
+// code for light (flashes when rover in autonomous mode, solid when manual)
+// light (code based on https://www.arduino.cc/en/Tutorial/BuiltInExamples/BlinkWithoutDelay)
+boolean mode = 1;     // 1 -> autonomous mode (light flashes), 0 -> manual mode (light solid)
+            // digital; placeholder pin number
+int light_state = LOW;                // used to set light
+unsigned long previousMillis = 0;   // stores last time light was updated
+const long interval = 1000;         // interval at which to blink (milliseconds)
+
+// motor speeds and commands (connect with ros)
+ros::NodeHandle nh;
+float right_speed = 0;
+float left_speed = 0;
+int right_dir = 0;
+int left_dir = 0;
+
+// get signed float speed command from ros, convert to magnitude and direction for right motor
+void rmotorCb(const std_msgs::Float64& control_msg){
+    last_ros_count = current_ros;
+    current_ros = millis();    
+    float right_input = control_msg.data;
+    if(right_input >= 0){
```

Comment:
Isn't this just bang-bang control? Do you think doing something more sophisicated is worth it ?

Reply:
This isn't really bang bang, this is just processing the motor control signals for the two wires (one analog for speed, one digital for direction). PID is implemented directly on the motor controller

Reply:
Got it